nemo/core/classes/mixins/adapter_mixins.py

```diff
@@ -159,6 +182,14 @@ def add_adapter(self, name: str, cfg: DictConfig):
             cfg: A DictConfig or Dataclass that contains at the bare minimum `__target__` to instantiate a
                 new Adapter module.
         """
+        if hasattr(self, '_accepted_adapter_types'):
+            if len(self._accepted_adapter_types) > 0:
+                test = model_utils.import_class_by_path(cfg._target_)
+                if test not in self._accepted_adapter_types:
+                    raise ValueError(
```

Comment:
Don't do a test like this. Create a bool, loop through the classes, do issubclass(test, cls_in_loop) and then if it's true, set the flag and break, check that the book is true, else raise value error.

Reply:
Whats the benefit of subclass checks? I'd rather check the exact class that is suppose to be added. I can imagine cases where the subclass behavior is very different from the parent, e.g. suppose we have MoE-Adapters subclassing Adapters.

Reply:
How about leaving it as exact check for now and if subclass is needed in the future we can revisit? @titu1994

Reply:
nvm, done as you suggested.